pgjdbc/src/main/java/org/postgresql/jdbc/PgArray.java

```diff
@@ -505,4 +506,21 @@ public void free() throws SQLException {
     fieldBytes = null;
     arrayList = null;
   }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
We should not implement equals/hash code for this class.
There was discussion on this in another issue or PR, but it will be a lot of work to get a true logical equivalence implementation, which just does not seem worthwhile.

Reply:
That's fine but I think that also brings up the question about how PgStruct equality should be handled.
PgStruct#resolveAttributes() will try to correct some casting issues between java objects and fix some time-related attributes. There are 3 scenarios in this case:
- remove Struct equality,
- keep Struct equality as is in the PR and let it fail if structs contain arrays,
- change equals implementation of PgStruct to iterate the attributes itself, and if it finds an Array, check their toString equality, (practically keeping PgArray equality isolated in PgStruct)

Personally, I think that keeping Struct equality would make it easier for the users as the other option would be that they go over the attributes themselves, but it also adds complexity in the driver which seems to be the same reason PgArray equality is not implemented.

Reply:
I think we should remove struct equality as well.